File: 20210222/LCS 2/lololalayoho.py
# 런타임 에러 발생
word1 = str(input())
word2 = str(input())
dp = [[0 for i in range(len(word1)+1)] for j in range(len(word2)+1)]
dp_word = [['' for i in range(len(word1)+1)] for j in range(len(word2)+1)]
for i in range(1,len(word1)+1):
    for j in range(1,len(word2)+1):
        if word1[i-1]==word2[j-1]:
            dp[i][j] = dp[i-1][i-1] + 1
            dp_word[i][j] = dp_word[i-1][j-1] + word2[j-1]
        else:
            if dp[i-1][j] < dp[i][j-1] :
                dp[i][j] = dp[i][j-1]
                dp_word[i][j] = dp_word[i][j-1]
            else:
                dp[i][j] = dp[i-1][j]
                dp_word[i][j] = dp_word[i-1][j]
print(dp[-1][-1])
print(dp_word[-1][-1])

File: 20210308/방문 길이/lololalayoho.java
import java.util.*;
class Solution {
    class positioin{
        int x;
        int y;
    }
    public positioin move(char dir){
        positioin p = new positioin();
        if(dir=='U'){
            p.x = -1;p.y = 0;
        }
        else if(dir=='D'){
            p.x = 1; p.y = 0;
        }
        else if(dir=='R'){
            p.x = 0; p.y = 1;
        }
        else if(dir=='L'){
            p.x = 0; p.y = -1;
        }
        return p;
    }
    public int solution(String dirs) {
        int answer = 0;
        List<String> roads = new ArrayList<>();
        positioin pos = new positioin();
        pos.x = 5; pos.y = 5;
        for(int i = 0; i<dirs.length();i++){
            positioin p = new positioin();
            p = move(dirs.charAt(i));
            int nextx = pos.x + p.x;
            int nexty = pos.y + p.y;
            if(nextx>=0 && nextx<11 && nexty>=0 && nexty <11){
                String word1 = String.valueOf(pos.x)+" "+String.valueOf(pos.y)+" "+String.valueOf(nextx)+" "+String.valueOf(nexty);
                String word2 = String.valueOf(nextx)+" "+String.valueOf(nexty)+" "+String.valueOf(pos.x)+" "+String.valueOf(pos.y);
                if(roads.contains(word1)==false && roads.contains(word2)==false){
                    answer = answer + 1;
                    roads.add(word1);
                    roads.add(word2);
                }
                pos.x = nextx; pos.y = nexty;
            }
        }
        return answer;
    }
}